packages/ec-core/src/operator/recombinator/mod.rs

```diff
@@ -3,25 +3,272 @@ use rand::rngs::ThreadRng;
 use super::{Composable, Operator};
 
 /// Recombine (usually two or more) genomes into a new
-/// genome.
+/// genome of the same type,
 ///
-/// GS: a generic for the _group_ of input genomes.
-/// Output: An associated type indicating what the result
-///   type of the recombination action will be.
+/// - `GS` represents the type for the _group_ of input genomes. This is
+///   typically a tuple, an array, or some other collection of genomes that will
+///   be recombined.
+/// - `Output`: An associated type indicating what the result type of the
+///   recombination action will be.
 ///
 /// Typically `Output` will be the same as the type of the
 /// individual genomes contained in `GS`, but that isn't captured
 /// (or required) here.
+///
+/// Implementations of this trait are typically representation dependent,
+/// so see crates like [`ec_linear`](../../../ec_linear/recombinator/index.html)
+/// for examples of recombinators on linear genomes.
+///
+/// # See also
+///
+/// See [`Recombine`] for a wrapper that converts a `Recombinator` into an
+/// [`Operator`], allowing recombinators to be used in chains of operators.
+///
+/// # Examples
+///
+/// In this example, we define a `Recombinator` that swaps one randomly
+/// chosen element from the first parent with the corresponding element
+/// from the second parent. We then use this recombinator to create a
+/// new genome from two parents, confirming that the child genome has
+/// all but one element from the first parent and one element from the
+/// second parent.
+///
+/// ```
+/// # use rand::thread_rng;
+/// # use rand::rngs::ThreadRng;
+/// # use rand::Rng;
+/// # use ec_core::operator::recombinator::Recombinator;
+/// #
+/// struct SwapOne;
+/// type Parents = (Vec<u8>, Vec<u8>);
```

Comment:
Vec?

Reply:
As I make this more generic (e.g., using `T` instead of `u8`), it seems less and less desirable to pin the collection down to a fixed length array. I thought this was a good change when I started it, but now I'm less clear.

Reply:
I would just use a fixed size array wherever possible and keep it as a vec where not, at least for test and example code tbh

Reply:
Generic makes sense for actual library code but not much for examples or tests

Reply:
For tests maybe not, but I think that _some_ (representative) genericity in examples could be helpful. If people are likely to copy examples, or at least base their code on examples, trying to illustrate "realistic" scenarios is a good thing as long as they're not so complicated that they obscure the more immediate point of the example.